### PR TITLE
[Kotlin] Update docs for concept exercise implementation

### PR DESCRIPTION
**Pull request by dector**
Mostly generic doc with custom file structure.

### CODE REVIEW

File: languages/kotlin/docs/implementing-a-concept-exercise.md [hunk 1]
```diff
@@ -1,5 +1,122 @@
...
-TODO: describe how to implement a concept exercise for the Kotlin track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes the steps required to implement a concept exercise in any v3 track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/docs/implementing-a-concept-exercise.md
+- `<exercise-slug>`: the name of the exercise in kebab-case (e.g. `functions-basic`).
+- `<ExerciseSlug>`: the name of the exercise in PascalCase (e.g. `FunctionsBasic`).
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the [exercise's GitHub issue](https://github.com/exercism/v3/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3Atrack%2Fkotlin+Implement+new+concept+exercise+in%3Atitle).
+
+Any concept exercise in any v3 track requires the following files to be created:
```
Comment (eparovyshnaya) [inline languages/kotlin/docs/implementing-a-concept-exercise.md]: There are lots of Kotlin-specifics in the following tree structure, which is not applicable to *any v3 track*. Do we really need to emphasise the intruction overwhelming nature here (and in the first sentence)?

File: languages/kotlin/docs/implementing-a-concept-exercise.md [hunk 2]
```diff
@@ -1,5 +1,122 @@
...
+
+- A stub implementation file (`src/main/kotlin/<ExerciseSlug>.kt`).
+- A file containing the test suite (`src/test/kotlin/<ExerciseSlug>.kt`).
+- An example implementation file that passes all the tests (`meta/Example.kt`).
+
+What these files look like depends on your track. Note that some tracks might require more files in addition to the three files just mentioned.
+
+## Step 8: add analyzer (optional)
+
+Some exercises could benefit from having an exercise-specific analyzer. If so, please check the track's analyzer document for details on how to do this.
```
Comment (eparovyshnaya) [inline languages/kotlin/docs/implementing-a-concept-exercise.md]: Can we provide here a link to the analyzer docs mentioned?

File: languages/kotlin/docs/implementing-a-concept-exercise.md [hunk 3]
```diff
@@ -1,5 +1,122 @@
...
+
+Skip this step if your track does not have an analyzer.
+
+## Step 9: custom representation (optional)
+
+Some exercises could benefit from having an custom representation as generated by the track's representer. If so, please check the track's representer document for details on how to do this.
```
Comment (eparovyshnaya) [inline languages/kotlin/docs/implementing-a-concept-exercise.md]: The same here. It's not obvious what is it all about and where to start. Can we reference something for further reading?

File: languages/kotlin/docs/implementing-a-concept-exercise.md (hunk 1 above)
Comment (dector) [reply languages/kotlin/docs/implementing-a-concept-exercise.md]: Sorry, didn't get your question. I think we'll try to use Kotlin-specific docs wherever we can and refer to generic ones for the rest.

File: languages/kotlin/docs/implementing-a-concept-exercise.md (hunk 2 above)
Comment (dector) [reply languages/kotlin/docs/implementing-a-concept-exercise.md]: I'd rather mark this as a draft as we don't have proper analyzer for now.

File: languages/kotlin/docs/implementing-a-concept-exercise.md (hunk 3 above)
Comment (dector) [reply languages/kotlin/docs/implementing-a-concept-exercise.md]: See response about analyzer. :)

**Comment by dector** [comment]
@eparovyshnaya added draft markers

File: languages/kotlin/docs/implementing-a-concept-exercise.md (hunk 1 above)
Comment (eparovyshnaya) [reply languages/kotlin/docs/implementing-a-concept-exercise.md]: My point is there is a plain textual discrepancy between this paragraph and the tree it explains: the tree cannot be applied to *any concept exercise in any v3 track*. The tree is Kotlin-specific.  It I were the reader, I'd prefer direct instruction that do not require deduction. 
It's ok for the first version though.

Comment (dector) [reply languages/kotlin/docs/implementing-a-concept-exercise.md]: I guess that's totally the point of this docs - to be language-specific (as they are in `languages/kotlin/` directory). :)

Previous paragraph also has link to kotlin-specific issues.

Comment (eparovyshnaya) [reply languages/kotlin/docs/implementing-a-concept-exercise.md]: And now, if we agreed on that the document should be entirely dedicated to *how to do it in Kotlin track* - shouldn't we stop insisting on *any exercise for any language* and *generic* nature if the document? What are your thoughts?

Comment (dector) [reply languages/kotlin/docs/implementing-a-concept-exercise.md]: Will DM you in Slack to avoid miscommunication. :)

**Pull request merged**